spanner/transaction.js

```diff
@@ -33,46 +33,37 @@ function readOnlyTransaction (instanceId, databaseId) {
 
   // Gets a transaction object that captures the database state
   // at a specific point in time
-  database.runTransaction({readOnly: true})
-    .then((results) => {
-      const transaction = results[0];
+  database.runTransaction({readOnly: true}, function(err, transaction) {
 
       const queryOne = 'SELECT SingerId, AlbumId, AlbumTitle FROM Albums';
 
       // Read #1, using SQL
-      transaction.run(queryOne)
-        .then((results) => {
-          const rows = results[0];
-
+      transaction.run(queryOne, function(err, rows) {
```

Comment:
Avoid callbacks and the `function` keyword - stick to promises and arrow functions where possible.